packages/protocol/contracts/signal/SignalService.sol

```diff
@@ -77,57 +78,69 @@ contract SignalService is ISignalService, EssentialContract {
     }
 
     /// @inheritdoc ISignalService
-    function isSignalReceived(
+    function proveSignalReceived(
         uint256 srcChainId,
         address app,
         bytes32 signal,
         bytes calldata proof
     )
         public
         view
-        validChainId(srcChainId)
-        validApp(app)
-        validSignal(signal)
         returns (bool)
     {
-        SignalProof memory signalProof = abi.decode(proof, (SignalProof));
-        bytes32 syncedSignalRoot = ICrossChainSync(resolve("taiko", false))
-            .getCrossChainSignalRoot(signalProof.height);
+        if (app == address(0)) return false;
```

Comment:
@adaki2004 Dani, this is the function that you need to carefully review. The multi-hop proof is supported in SignalService, not the Bridge.

Acutally it also make sense to remove `skipProofCheck` from the Bridge contract and add the function to SignalService.sol, but when I tried to do it, quite a few tests failed. Maybe you can give it a try.

Reply:
> @adaki2004 Dani, this is the function that you need to carefully review. The multi-hop proof is supported in SignalService, not the Bridge.
> 
> Acutally it also make sense to remove `skipProofCheck` from the Bridge contract and add the function to SignalService.sol, but when I tried to do it, quite a few tests failed. Maybe you can give it a try.

Done.
You had also 2 questions (ERC721 and ERC1155 tests related)  and i'm reflecting here:
```
// TODO(dani): we have to overwrite Alice address, otherise
// test_onMessageRecalled_721 will fail. Do you know why?
```
```
// TODO(dani): why chaning Amilia's address will fail the test?
```

Because we need to create the `getPreDeterminedDataBytes()` pre-testing. (We need to gain this data from the source chain’s `sendToken()`’s `message.data` but in order to do that i need to pre-calculate it (off-the-official-tests), locally.
 This will bake in the `msg.sender `context, which is Alice/Amelia (in ERC721 it was former, otherwise latter). So this is why it fails, because it affects the predetermined data bytes. 
 Now I modified and Amelia is the caller, so a bit unified the 2 tests, but the reason was that - to not change them.